lib/view.js

```diff
@@ -0,0 +1,73 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-connector-cloudant
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var URL = require('url');
+var assert = require('assert');
+var util = require('util');
+var _ = require('lodash');
+
+module.exports = mixinView;
+
+function mixinView(Cloudant) {
+  var debug = require('debug')('loopback:connector:cloudant:view');
+
+  /**
+   * Gets data at `/{db}/_design/{ddocName}/views/{viewName}`
+   *
+   * Example:
+   * User has a view called `getModel` in design document /{db}/_design/model,
+   * to query the view, user can call function:
+   * ```
+   * ds.viewDocs(model, getModel, {key: 'purchase'}, cb);
+   * ```
+   *
+   * @param {String} ddocName The design doc name
+   * @param {String} viewName The view name
+   * @param {Object} options The cloudant view filter
+   * @callback
+   * @param {Function} cb
+   */
+  Cloudant.prototype.viewDocs = function(ddocName, viewName, options, cb) {
+    // omit options, e.g. ds.viewDocs(ddocName, viewName, cb);
+    if (typeof options === 'function' && !cb) {
+      cb = options;
+      options = {};
+    }
+    debug('Cloudant.prototype.view ddocName %s viewName %s options %s',
+      ddocName, viewName, options);
+
+    var self = this;
+    var db = this.cloudant.use(self.getDbName(self));
+    assert(db && typeof db.view === 'function',
```

Comment:
This isn't checking the design doc or the datasource. `this.cloudant.use` is just returning a scoped object so that methods on it use URLs based on the db name. If `db.view` is ever not a function then it would mean that someone has changed the implementation of this connector to no longer use `cloudant`/`cloudant-nano`/`nano` as the client.

Is that the purpose of this assertion?